SchedulerBot/SchedulerBot.Business.Commands/AddCommand.cs

```diff
@@ -45,40 +42,32 @@ public class AddCommand : IBotCommand
 			SchedulerBotContext context,
 			IScheduleParser scheduleParser,
 			IScheduleDescriptionFormatter scheduleDescriptionFormatter,
-			ILogger<AddCommand> logger)
+			ILogger<AddCommand> logger) : base("add", logger)
 		{
 			this.context = context;
 			this.scheduleParser = scheduleParser;
 			this.scheduleDescriptionFormatter = scheduleDescriptionFormatter;
-			this.logger = logger;
-
-			Name = "add";
 		}
 
 		#endregion
 
-		#region IBotCommand Implementation
-
-		/// <inheritdoc />
-		public string Name { get; }
+		#region Overrides
 
 		/// <inheritdoc />
-		public async Task<CommandExecutionResult> ExecuteAsync(Activity activity, string arguments)
+		protected override async Task<CommandExecutionResult> ExecuteCoreAsync(Activity activity, string arguments)
```

Comment:
Possibly need to change naming to remove `ExecuteCoreAsync` -> `ExecuteCoreLogicAsync`?

Reply:
Not sure about that, it is a pretty common pattern for naming an abstract method called from another one:

https://msdn.microsoft.com/en-us/library/system.web.mvc.controller.executecore(v=vs.118).aspx
https://msdn.microsoft.com/en-us/library/system.windows.uielement.measurecore(v=vs.110).aspx

So I'd leave it as it is.